include/libdnf5/repo/repo_sack.hpp

```diff
@@ -165,6 +165,7 @@ class RepoSack : public sack::Sack<Repo> {
 
     repo::Repo * system_repo{nullptr};
     repo::Repo * cmdline_repo{nullptr};
+    bool repos_updated_and_loaded{false};
```

Comment:
OK, just in this file also please mark the `update_and_load_repos` method as experimental, using f.e. the @warning tag as described in my https://github.com/rpm-software-management/dnf5/pull/680#issuecomment-1665187488.